### PR TITLE
introduce xoclv2 to XRT utilities

### PR DESCRIPTION
**Pull request by maxzhen**
This change allows utilities to discover devices driven by xoclv2 driver.

### CODE REVIEW

File: src/runtime_src/core/pcie/linux/scan.h [hunk 1]
```diff
@@ -81,10 +102,12 @@ class pci_device
   std::string sysfs_name =    ""; // dir name under /sys/bus/pci/devices
   int user_bar =              0;  // BAR mapped in by tools, default is BAR0
   size_t user_bar_size =      0;
-  bool is_mgmt =              false;
+  enum pf_type pf_type =      XCLMGMT;
+  bool is_v2_drv() { return pf_type == XMGMT || pf_type == XUSER; }
```
Comment (sonals) [inline src/runtime_src/core/pcie/linux/scan.h:106]: `bool is_v2_drv() const { return pf_type == XMGMT || pf_type == XUSER; }`

Const method

File: src/runtime_src/core/pcie/linux/scan.h [hunk 2]
```diff
@@ -81,10 +102,12 @@ class pci_device
...
+  bool is_mgmt() { return pf_type == XCLMGMT || pf_type == XMGMT; }
```
Comment (sonals) [inline src/runtime_src/core/pcie/linux/scan.h:107]: const method

File: src/runtime_src/core/pcie/linux/scan.h [hunk 3]
```diff
@@ -63,6 +63,27 @@ struct fdt_header {
...
 
+enum pf_type {
+    UNKNOWN,
+    XCLMGMT,
+    XMGMT,
+    XOCL,
+    XUSER
+};
+
+static inline enum pf_type drv2type(const std::string driver)
```
Comment (sonals) [inline src/runtime_src/core/pcie/linux/scan.h:74]: `static inline enum pf_type drv2type(const std::string &driver)`

const reference

File: src/runtime_src/core/pcie/linux/scan.h [hunk 4]
```diff
@@ -63,6 +63,27 @@ struct fdt_header {
 
 namespace pcidev {
 
+enum pf_type {
+    UNKNOWN,
+    XCLMGMT,
+    XMGMT,
+    XOCL,
+    XUSER
+};
```
Comment (sonals) [inline src/runtime_src/core/pcie/linux/scan.h:72]: Like this concept of PF type.

File: src/runtime_src/core/pcie/linux/scan.cpp [hunk 5]
```diff
@@ -155,12 +170,17 @@ get_render_value(const std::string& dir)
 }
 
 static bool
-devfs_exists(bool is_mgmt, uint32_t instance)
+devfs_exists(pcidev::pf_type type, uint32_t instance, std::string& sysfsname)
 {
   struct stat buf;
-  std::string devfs = get_devfs_path(is_mgmt, instance);
+  std::string devfs = get_devfs_path(type, instance, sysfsname);
```
Comment (sonals) [inline src/runtime_src/core/pcie/linux/scan.cpp:176]: `const std::string devfs = get_devfs_path(type, instance, sysfsname); `

Comment (maxzhen) [reply src/runtime_src/core/pcie/linux/scan.cpp]: fixed.

File: src/runtime_src/core/pcie/linux/scan.h (hunk 4 above)
Comment (maxzhen) [reply src/runtime_src/core/pcie/linux/scan.h:72]: fixed.

File: src/runtime_src/core/pcie/linux/scan.h (hunk 3 above)
Comment (maxzhen) [reply src/runtime_src/core/pcie/linux/scan.h]: fixed.

File: src/runtime_src/core/pcie/linux/scan.h (hunk 2 above)
Comment (maxzhen) [reply src/runtime_src/core/pcie/linux/scan.h]: fixed.

File: src/runtime_src/core/pcie/linux/scan.h (hunk 1 above)
Comment (maxzhen) [reply src/runtime_src/core/pcie/linux/scan.h]: fixed.

File: src/runtime_src/core/pcie/linux/scan.cpp [hunk 6]
```diff
@@ -150,12 +165,17 @@ get_render_value(const std::string& dir)
...
   struct stat buf;
-  std::string devfs = get_devfs_path(is_mgmt, instance);
+  const std::string devfs = get_devfs_path(type, instance, sysfsname);
+
+  if (stat(devfs.c_str(), &buf) == 0)
+    return true;
 
-  return (stat(devfs.c_str(), &buf) == 0);
+  // For xoclv2 golden image, there will only be a flash devfs node
+  std::string flash_devfs = "/dev/xfpga/flash." + sysfsname;
```
Comment (sonals) [inline src/runtime_src/core/pcie/linux/scan.cpp:177]: const

File: src/runtime_src/core/pcie/linux/scan.cpp [hunk 7]
```diff
@@ -792,32 +805,23 @@ class pci_device_scanner
   }
 
 private:
-  void rescan_nolock()
+  void rescan_nolock(const std::string driver)
   {
-    if (is_in_use(user_list) || is_in_use(mgmt_list)) {
-      std::cout << "Device list is in use, can't rescan" << std::endl;
+    std::string drvpath = sysfs::drv_root + driver;
```
Comment (sonals) [inline src/runtime_src/core/pcie/linux/scan.cpp:810]: `const`

**Pull request merged**